### DISCUSSION TITLE
Marlin Firmware?

### DISCUSSION THREAD

**Discussion by bonz099** [xpDIY/3DPlacer#7] [General]
Im very new to OPNP, so this may be a config issue on my end, but i cant get my gcode driver to connect. I keep getting this message. 

Marlin firmware is not reporting support for rotation axes (A B C). FIRMWARE_NAME:Marlin 1.1.9.1 (Github) SOURCE_CODE_URL:https://github.com/MarlinFirmware/Marlin PROTOCOL_VERSION:1.0 MACHINE_TYPE:Bonz Opnp 1.0 EXTRUDER_COUNT:1 UUID:cede2a2f-41a2-4748-9b12-c55c62f367ff

Was there any other changes you made to your firmware to get Opnp to connect?

**Comment by bonz099** [comment]
Never mind, I guess I can go through the setup ignoring that.

**Reply by xpDIY** [reply]
I also had that, so I ignore that and use E axis to drive the rotation, also in OPnp, for axis, there is an option linear->rotation, when you see your rotation is very slow, you can enable this option. 

For marlin firmware, I don't need to make any change, some of the firmware when press connect button, it will keep reseting, in that case, you can change the connect command in G-Code driver to empty.

**Reply by bonz099** [reply]
How do you configure the Rotation axis, what is the gear reduction setup as? Could you post a screenshot of your config?

**Reply by xpDIY** [reply]
first you should configure the axis letter, like this,
![image](https://github.com/xpDIY/3DPlacer/assets/120340363/13930255-e2eb-4c91-b8e4-2017064ab08c)
then you need to configure mapping of that axis to let it calibrate to 360 degree, like this,
![image](https://github.com/xpDIY/3DPlacer/assets/120340363/cec82ecc-6433-4563-a9ad-1013a80ca50c)
the value will vary for different printer, for me it is 16.2 map to 360 degree, you can mark the gears, and then send G command (G1 Exx) 
to let it turn 360 degree, then use that value as mapping setting.

**Reply by bonz099** [reply]
Im not sure I follow, what is the Gcode command that should should use as reference for a full 360. Currently my value is 35, but ide love to understand better how to fine tune it, since my PNP is nearly working working now. Gears are marked, just need to fine tune the rotation mapping now.

**Reply by xpDIY** [reply]
So you can do first G1 E0, then mark the location of your gear, then you do G1 E35, then see if the nozzle has rotated 360 degree, note it is the nozzle but not the stepper motor who should turn 360. If it is too much, you can reduce a bit, for ex. G1 E34.9, after the mark turn full 360 degree, then you can put the value 34.9 for ex. to Map Point B input. Seems like I mistype the Gcode in above comment, so I corrected it.

**Comment by bonz099** [comment]
AWESOME! Thank you! I cant wait to work on this some more this weekend.

**Comment by bonz099** [comment]
I just got my machine working over the weekend, thank you!

**Reply by xpDIY** [reply]
Super! Enjoy 👍
